### PR TITLE
Enable BWC checks

### PR DESCRIPTION
**Pull request by adnapibar**
### Description
This PR enables the BWC checks in OpenSearch. The existing issues were fixed by the following PRs
- #764 
- #708 
- #775
- #797 

### Issues Resolved
The following issues can be resolved.

- #669 
- #417 
- #113 

### Check List
~- [ ] New functionality includes testing.~
  - [ ] All tests pass
~- [ ] New functionality has been documented.~
~- [ ] New functionality has javadoc added~
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Rabi Panda <adnapibar@gmail.com>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -183,9 +183,9 @@ tasks.register("verifyVersions") {
  * after the backport of the backcompat code is complete.
  */
 
-boolean bwc_tests_enabled = false
```
Comment (dblock) [inline build.gradle:186]: Should we just kill this part. Seems like a relic of breaking backwards compatibility tests, no?

Comment (adnapibar) [reply build.gradle:186]: I think we might need it to disable the bwc checks temporarily from time to time as explained in the comment above.

Comment (dblock) [reply build.gradle:186]: Well, it doesn't explain why. Why?

**Review by harold-wang — APPROVED** [review]
looks good, thanks Rabi!

File: build.gradle (hunk 1 above)
Comment (adnapibar) [reply build.gradle:186]: Hmm, when it says "necessary", I think, if you don't disable it then until you backport the change to other major release branches, the checks will start failing in CI and that will stop you merging the code to main?

Maybe, let's just keep it for now and if we think this is not necessary we will remove it later.

**Review by dblock — APPROVED** [review]

**Comment by adnapibar** [comment]
Waiting on #797 to be merged before starting `gradle check` and subsequently merging it.

**Comment by dblock** [comment]
start gradle check

**Comment by adnapibar** [comment]
start gradle check

**Comment by dblock** [comment]
@adnapibar Sounds like a merge to me!

**Pull request merged**